easybuild/tools/filetools.py

```diff
@@ -763,6 +763,13 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None):
     apatch = os.path.abspath(patch_file)
     adest = os.path.abspath(dest)
 
+    # Attempt extracting the patch if it does not end in .patch (may end in .gz, .zip, etc.)
+    if not apatch.endswith(".patch"):
```

Comment:
Ok. In the latest commit, it has to end in ".patch.gz", ".patch.bz2" or ".patch.xz". 

I don't think we can assume other formats are not an archive with a bunch of directories in it.